### ISSUE TITLE
pullstream: 'callback' is not a function at _flush()

### ISSUE THREAD

**Issue by popeindustries** [EvanOxfeld/node-unzip#19]
Getting an error after updating to Node 0.10.0:

``` bash
TypeError: undefined is not a function
    at Until.PullStream._flush (/Users/n23618/PROJECTS/buddy-dependencies/node_modules/unzip/node_modules/pullstream/pullstream.js:112:5)
    at Until.<anonymous> (_stream_transform.js:131:12)
    at Until.g (events.js:175:14)
    at Until.EventEmitter.emit (events.js:117:20)
    at finishMaybe (_stream_writable.js:332:12)
    at endWritable (_stream_writable.js:339:3)
    at Until.Writable.end (_stream_writable.js:326:5)
    at /Users/n23618/PROJECTS/buddy-dependencies/node_modules/unzip/lib/parse.js:277:36
    at process._tickCallback (node.js:415:13)
```

Code:

``` javascript
request.get(this.url)
  .pipe(fs.createWriteStream(filename))
  // Error downloading
  .on('error', function(err) {
    fn('fetching ' + self.url + ' failed: ' + err);
  })
  .on('close', function() {
    extractor = unzip.Extract({path: self.temp});
    writer = fstream.Writer(self.temp);
    writer.on('close', fn);
    // Unzip
    fs.createReadStream(filename)
      .pipe(unzip.Parse())
      // Store path to unzipped package
      .on('entry', function(entry) {
        if (!self.location && entry.type === 'Directory') self.location = path.resolve(self.temp, entry.path);
      })
      // Error unzipping
      .on('error', function() {
        fn('unzipping archive: ' + filename);
      })
      .pipe(writer);
  });
```

**Comment by EvanOxfeld** [comment]
Thanks for reporting the issue - at least one of the unzip dependencies (pullstream) needs to be updated for node v0.10.0. I'll publish a new version of pullstream later today. In the meantime, you're welcome to try out the match-wip branch (currently on commit 58e006b61be4f258eb4c1164cdc6d2c3f7771807).

**Comment by timsweb** [comment]
My stab at a fix: nearinfinity/node-pullstream#5.

**Comment by popeindustries** [comment]
It looks like pullstream 0.2.1 fixes the problem. Thanks.

**Comment by reinseth** [comment]
Am I missing something here? Why is this issue closed when the pullstream dependency is still listed at ~0.1.0?

**Comment by popeindustries** [comment]
@reinseth you're right. Wrong button. Reopening until the latest version of pullstream has been added.

**Comment by EvanOxfeld** [comment]
Even with this issue resolved, unzip still has at least one more major Node 0.10 compatibility problem to fix - see #20 for problems inflating compressed files larger than 16 kb.

Right now the plan is to wait until #20 is fixed before publishing a new version to NPM, but I'm open to publishing an alpha release if it would be useful. Thanks again for your help.

**Comment by shama** [comment]
+1 for an alpha release. Thanks for the quick fix on this btw!

**Comment by cowboy** [comment]
If a half-fix release would be more fixed than the current version, I'm +1.

**Comment by aaronallport** [comment]
+1 for an alpha release

**Comment by EvanOxfeld** [comment]
Thanks for the input everyone. The fix for this issue landed in `0.1.6-alpha` on npm.

**Comment by shama** [comment]
Thank you @EvanOxfeld!
